Sort leaderboard by net worth instead of balance

rickles42 opened the pull request:
Currently, the leaderboard shows the users with the highest balances,
but this is probably not the best indicator of our top investors. The
top investors are the ones with the highest net worth, i.e. balance plus
value of active investments. Also updates the !top command to match.

Fixes #145

rickles42 committed:
Sort leaderboard by net worth instead of balance

Currently, the leaderboard shows the users with the highest balances,
but this is probably not the best indicator of our top investors. The
top investors are the ones with the highest net worth, i.e. balance plus
value of active investments. Also updates the !top command to match.

rickles42 commented:
@robalb FYI this touches some of the `main.js` and `index.html` code, so if you touch similar lines in your rework, we may have to resolve a merge conflict when we merge your PR.

Also changes how the `/top` API works (now it always returns highest net worth investors). Let me know if you were using any of the other functionality.

robalb commented:
Ok, but do we really need to show the balance in the leaderboard?

We could show only the net worth, and put the rest in the user page, with all the other user data like the active investments

rickles42 commented:
@robalb yeah if you look at my change you'll see the leaderboard now shows net worth instead of balance

robalb commented:
Lol sorry, for some reason i didn't notice the deletion before the addition when i looked at it.

In my defence, i had just woken up

rickles42 commented:
Haha no problem

thecsw commented:
Awesome! Several questions, what does index mean here?
``` python
name = Column(String(20), nullable=False, index=True)
```

Yeah, everything seems good!

rickles42 commented:
In SQL databases, columns can be optionally indexed. For an indexed column, the database generates a bunch of extra data internally (like hash tables and stuff) that makes it much much faster to read and sort values in that column. The trade-off is that values in that column are a little slower to write or update (since the hash tables and stuff need to be updated too).

The reason I added this is that, without it, the JOIN operation in my SQL query took ~30 seconds. That's because JOINing on the `name` column requires lots and lots of reads and searches in that column. When I added the index, the operation dropped to ~0.2 seconds. Writes still seem to be fine.
